llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -6710,13 +6739,21 @@ void SelectionDAGBuilder::visitIntrinsicCall(const CallInst &I,
       return;
     }
 
-    // assert if Payloads type are Vector
-    assert(!VT.isVector() &&
-           "Vector type not supported yet for fallback implementation");
+    // We don't support non-integral pointers
+    Type *CurrType = VT.getTypeForEVT(*Context);
+    if (CurrType->isPointerTy()) {
+      unsigned AS = CurrType->getPointerAddressSpace();
+      if (DAG.getDataLayout().isNonIntegralAddressSpace(AS)) {
+        report_fatal_error(
+            "llvm.ct.select: non-integral pointers are not supported");
+      }
+    }
 
-    // We don't support floating points yet
-    assert(!VT.isFloatingPoint() &&
-           "Float point type not supported yet fallback implementation");
+    // We don't support scalable vector types yet, for now it'll only be
+    // fix-width vector
+    // TODO: Add support for scalable vectors
+    assert(!VT.isScalableVector() &&
```

Comment:
Same here. We should add llvm unreachable or fatal error to avoid issues when assert is disabled.